test/powershell/Modules/Microsoft.PowerShell.Management/Clipboard.Tests.ps1

```diff
@@ -41,5 +41,15 @@ Describe 'Clipboard cmdlet tests' -Tag CI {
             'world' | Set-Clipboard -Append
             Get-Clipboard -Raw | Should -BeExactly "hello$([Environment]::NewLine)world"
         }
+
+        It 'Set-Clipboard -PassThru returns single object' {
+            $result = 'hello' | Set-Clipboard -PassThru
+            $result | Should -BeExactly 'hello'
+        }
+
+        It 'Set-Clipboard -PassThru returns multiple objects' {
+            $result = 'hello', 'world' | Set-Clipboard -PassThru
```

Comment:
seems like the return value should be what is in the clipboard

Reply:
Changed to use `Get-Clipboard`.

Reply:
I may have misunderstood. You mean it should return the current clipboard instead of just passing through the current items correct? If so, what should the behavior be as `Get-Clipboard` outputs as a string array by default and single string when `-Raw` is used.

Reply:
The test description states "PassThru returns". So I think original test was good.

Reply:
My comment still applies, -PassThru should return what's actually put in the clipboard, which in this case won't be an array.

Reply:
Would a user expect `-PassThru` to output everything in the clipboard or just the lines that was passed into that instance of the command?

```powershell
'Item1' | Set-Clipboard
'Item2' | Set-Clipboard -Append -PassThru

# Should the return be?

Item1
Item2

# Or

Item2
```

I personally think the user would expect that only what was passed to `Set-Clipboard -PassThru` would be returned, not the entire clipboard. The user may want to only use the specific item passed to `Set-Clipboard` and not the entire clipboard. If the user wanted the entire clipboard they would use `Get-Clipboard` instead.

Reply:
Current Set-Clipboard behavior is to accumulate values until user clean clipboard explicitly. It is true for pipeline scenario too.
```
'hello', 'world' | Set-Clipboard
Get-Clipboard

hello
world
```
 Why would the PassThru change the behavior? If we want to support new scenario we could add new switch to clean clipboard before put new value to it.

Reply:
@iSazonov your second example is not what I'm describing. This is the correct output:

```powershell
$result = 'hello', 'world' | Set-Clipboard -PassThru 
$result

hello
world

Get-Clipboard

hello
world
```

`Set-Clipboard` without the `-Append` switch parameter will always contain and return the exact data passed to it. The scenario I'm talking about is when the clipboard already has data and the `-Append` switch is used to add data. In that scenario should be behavior be to output the entire clipboard or just the the passed data.

### Return entire clipboard

```powershell
'Item1' | Set-Clipboard
'Item2' | Set-Clipboard -Append -PassThru

Item1
Item2
```

### Return only passed data

```powershell
'Item1' | Set-Clipboard
'Item2' | Set-Clipboard -Append -PassThru

Item2
```

Reply:
As for Append, I agree the cmdlet shouldn't return more than it get from pipeline. Otherwise it would again weird behavior.

Reply:
We should have this discussion in the issue instead of the PR to get more feedback from users and not just developers.